libcodechecker/server/client_db_access_handler.py

```diff
@@ -758,12 +787,30 @@ def getRunResultCount(self, run_ids, report_filter, cmp_data):
                     # There is no difference.
                     return 0
 
-            q = session.query(Report) \
-                .filter(Report.run_id.in_(run_ids)) \
-                .outerjoin(File, Report.file_id == File.id) \
-                .outerjoin(ReviewStatus,
-                           ReviewStatus.bug_hash == Report.bug_id) \
-                .filter(filter_expression)
+            isUnique = report_filter is not None and \
```

Comment:
This code is very similar to the one in `getRunResults`. Maybe it makes sense to hoist some of the functionality into functions to reduce the duplication?